src/text-buffer.coffee

```diff
@@ -13,6 +13,40 @@ MarkerStore = require './marker-store'
 Patch = require './patch'
 {spliceArray, newlineRegex} = require './helpers'
 
+class SearchResultCallback
```

Comment:
> Not sure about the name "callback" as it implies a function to me, whereas this is an object.

Agreed. :+1: 

> Or are you looking for some other word to indicate the control methods it provides?

Exactly. I'd expect a `SearchResult` to be a simple value object, whereas this one provides a bunch of features which seem to be outside the scope of an object which merely holds data.

Also, I would stray a bit from `SearchResult` as it may imply that we return the result of the whole search, whilst what we're really passing along is a *search occurrence*.

I am thinking to something along the lines of `SearchOccurrence`, but that clearly expresses that it provides control over the iteration as well. `SearchOccurrenceIteratee` maybe? :confused: